lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py

```diff
@@ -348,12 +348,17 @@ def _attached_sds_service(self, dc_name):
 
     def _attached_sd_service(self, storage_domain):
         dc_name = self._module.params['data_center']
-        # Find the DC, where the storage we want to remove reside:
-        if not dc_name and self._module.params['state'] == 'absent':
+        # Find the DC, where the storage resides:
+        if not dc_name:
             dcs_service = self._connection.system_service().data_centers_service()
             dc = search_by_attributes(dcs_service, storage=storage_domain.name, status='up')
             if dc is None:
-                raise Exception("Can't remove storage, because no active datacenter found for it's removal")
+                raise Exception(
+                    "Can't bring storage to state `%s`, because it seems that"
+                    "it is not attached to any active datacenter"
```

Comment:
I would think to rephrase the message since it suggests that the storage domain is not attached although we search for an active storage, but storage domain can also be in maintenance mode or inactive and still be attached to a DC.
maybe something like "Can't bring storage to state `%s`, because no active storage domain found in DC"

Reply:
"Can't bring storage to state %s, because no active storage domain found in DC" -> which DC? This is that I am searching for...

With this search I am really trying to find the datacenter in which the SD is attached.
If it's in maintenance it's still attached as you said.

Reply:
I see.
If you desire to find the datacenter, only for the log, which the SD is attached to , then you should search the DC without the filtering of the status, as so:
   dc = search_by_attributes(dcs_service, storage=storage_domain.name)
Then if the dc is empty then the storage domain is detached and if the dc is not empty, but we got an empty dc with the latest fetch (with DC status = 'up')  then we can say that the storage domain is attached to a DC although it is not up.

Alternatively, you can fetch the DC and check its status, the same logic which I explained before should apply here also.

Reply:
`If you desire to find the datacenter, only for the log,` >> I use it afterwards, it's not only for the logs!

@maorlipchuk 
* Notice that in case the `state` parameter is `unattached` or `absent` and the storage is already in detached state we will never reach that part of code (_attached_sd_service) because of pre-remove step.
* In case the `state` parameter is `maintenance`, and SD state is `unattached`, I guess the module should fail, like it does now. 
*  I don't get this sentence. `we can say that the storage domain is attached to a DC although it is not up.` What is not UP.

So, can you please re-explain what changes you would like that I make there?

Thanks a lot!

Reply:
> I don't get this sentence. we can say that the storage domain is attached to a DC although it > is not up. What is not UP.

The search for the DC is being done using the following search:
    dc = search_by_attributes(dcs_service, storage=storage_domain.name, status='up')
 status='up' - We filter the data centers with 'up' status, so you might get an empty DC although oVirt will contain a DC with attached storage domain but not in 'up' status

Since you mention that you will use the DC afterwards I suggest to fetch DC without the status filter:
   dc = search_by_attributes(dcs_service, storage=storage_domain.name)
and check if the dc is with 'up' status or not, if it isn't 'up' then we can raise an exception

Reply:
With this change you can now mention In the exception the DC name, as so so:
....
  dc = search_by_attributes(dcs_service, storage=storage_domain.name)
  if dc is None:
                 raise Exception(
                     "Can't bring storage to state `%s`, because it seems that"
                     "it is not attached to any active datacenter"
else:
    "Can't bring storage to state %s, because no active storage domain found in DC %s"

Reply:
I think you forgot some ifs in the code snippet but I get what you want to say, reposting.